### PR TITLE
Issue 6268 Remove shared editor libraries

### LINKED ISSUE #6268: Remove shared libraries for the unsupported desktop platforms
Currently, we include shared libraries for all 3 desktop platforms in the editor bundle.
E.g. in the macOS editor, I see the .so and .dll files for javafx.

The size for each platform is between 70-110mb.

### PR DESCRIPTION
**Pull request by britzl**
Fixes #6268

### CODE REVIEW

**Comment by britzl** [comment]
Resulting app size after change: 303.9MB

File: editor/scripts/bundle.py [hunk 1]
```diff
@@ -312,6 +312,41 @@ def get_exe_suffix(platform):
     return ".exe" if 'win32' in platform else ""
 
 
+def strip_platform_files(platform, jar):
```
Comment (JCash) [inline editor/scripts/bundle.py:315]: I prefer "remove", or "strip_platform_files_from_archive", since "strip" is another actual term that we use often.

**Comment by AGulev** [comment]
👍
We will save a few more megabytes for maos/linux when update javafx for the next time: https://github.com/defold/defold/issues/6268#issuecomment-999882062

File: editor/scripts/bundle.py [hunk 2]
```diff
@@ -312,6 +312,41 @@ def get_exe_suffix(platform):
...
+
+    # find libs to remove in the root folder
+    for file in files:
+        if "/" not in file:
+            if platform == "x86_64-darwin" and (file.endswith(".so") or file.endswith(".dll")):
+                files_to_remove.append(file)
+            elif platform == "x86_64-win32" and (file.endswith(".so") or file.endswith(".dylib")):
+                files_to_remove.append(file)
+            elif platform == "x86_64-linux" and (file.endswith(".dll") or file.endswith(".dylib")):
+                files_to_remove.append(file)
```
Comment (JCash) [inline editor/scripts/bundle.py:334]: Howcome the files were included in the first place?
Shouldn't the fix be there?

**Review by JCash — DISMISSED** [review]

File: editor/scripts/bundle.py (hunk 2 above)
Comment (AGulev) [reply editor/scripts/bundle.py:334]: I think it's specified here: 
https://github.com/defold/defold/blob/c5a77bd14529894bccd84fa73264111bbca4c4aa/editor/project.clj#L81-L119

not sure how to specify only needed libs depends on platform there

Comment (britzl) [reply editor/scripts/bundle.py:334]: We take everything from bob and the editor and put it into editor/target/defold-editor-2.0.0-SNAPSHOT-standalone.jar. We then use this jar file for all editor builds regardless of platform. So the first shot at something platform specific is when we actually bundle the editor for a specific platform.

We could obviously create on editor/target/defold-editor-2.0.0-SNAPSHOT-standalone.jar per platform if we want to, but I don't see that as better than the current solution.

Comment (britzl) [reply editor/scripts/bundle.py:334]: > I think it's specified here

You're right. We might be able to exclude something there.

**Pull request merged**